src/physics/physics.cpp

```diff
@@ -164,12 +164,24 @@ void solve_collision(Manifold hit, real delta) {
     }
 }
 
+Manifold PhysicsEngine::hitscan(Vec3 origin, Vec3 direction, EntityID sender) {
+    for (auto &body : bodies) {
+        if (body.entity == sender) continue;
+        Manifold hit = collision_line_aabody(origin, direction, &body);
+        if (hit) {
+            hit.a = &body;
+            return hit;
+        }
+    }
+    return {};
+}
```

Comment:
The empty `Manifold` is truthy:
```cpp
struct Manifold {
    real t;
    real depth;
    // ...
    operator bool() const {
        // We include 0, since collisions when the origin
        // is inside the box exists.
        return (t > 0 && t <= 1) || (depth >= 0 && t >= 0);
    }
};
```
Do we want to return something else here?

Reply:
The comment raises a good point, and it  actually makes me question the decision to return `{}` from hitscan. I think `{ -1 }` is more inline with what `collision_line_aabody` and other collision functions generate.